azure-iot-device/tests/iothub/test_sync_clients.py

```diff
@@ -99,59 +111,98 @@ def test_auth_provider_creation(self, mocker, client_class, connection_string, c
     @pytest.mark.parametrize(
         "ca_cert",
         [
-            pytest.param(None, id="No CA certificate"),
-            pytest.param("some-certificate", id="With CA certificate"),
+            pytest.param(None, id=" No CA certificate"),
+            pytest.param("some-certificate", id=" With CA certificate"),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "websockets",
+        [
+            pytest.param(None, id=" Default (No Websockets) "),
+            pytest.param(False, id=" No Websockets "),
+            pytest.param(True, id=" With Websockets "),
         ],
     )
     def test_pipeline_creation(
-        self, mocker, client_class, connection_string, ca_cert, mock_pipeline_init
+        self, mocker, client_class, connection_string, ca_cert, websockets, mock_pipeline_init
     ):
         mock_auth = mocker.patch(
             "azure.iot.device.iothub.auth.SymmetricKeyAuthenticationProvider"
         ).parse.return_value
 
+        mock_config_init = mocker.patch(
+            "azure.iot.device.iothub.abstract_clients.config.BasePipelineConfig"
+        )
+
         args = (connection_string,)
         kwargs = {}
         if ca_cert:
             kwargs["ca_cert"] = ca_cert
+        if websockets:
+            kwargs["websockets"] = websockets
         client_class.create_from_connection_string(*args, **kwargs)
-
+        assert mock_config_init.call_count == 1
+        if websockets:
+            assert mock_config_init.call_args == ({"websockets": websockets},)
```

Comment:
>assert mock_config_init.call_args == ({"websockets": websockets},) [](start = 12, length = 66)

I'm not sure if this is significant or not, but I think it might be more appropriate to use mocker.call(), as in: `assert mock_config_init.call_args == mocker.call(websockets=websockets)`

Reply:
so, I think mocker call actually changes things? I looked this up, because originally I did `mocker.call()` but the test failed. Looked up the documentation and they explained it nicely:

```
>>> mock = Mock(return_value=None)
>>> print(mock.call_args)
None
>>> mock()
>>> mock.call_args
call()
>>> mock.call_args == ()
True
>>> mock(3, 4)
>>> mock.call_args
call(3, 4)
>>> mock.call_args == ((3, 4),)
True
>>> mock.call_args.args
(3, 4)
>>> mock.call_args.kwargs
{}
>>> mock(3, 4, 5, key='fish', next='w00t!')
>>> mock.call_args
call(3, 4, 5, key='fish', next='w00t!')
>>> mock.call_args.args
(3, 4, 5)
>>> mock.call_args.kwargs
{'key': 'fish', 'next': 'w00t!'}
```

Reply:
+1

---
In reply to: [335151026](https://github.com/Azure/azure-iot-sdk-python/pull/331#discussion_r335151026) [](ancestors = 335151026)